### DISCUSSION TITLE
How to include modules assemblies without reference

### DISCUSSION THREAD

**Discussion by alfred-zaki** [dotnet/aspnetcore#28979] [Q&A]
Hi,

previously i was able to achieve plug-in like architect buy build my modules dlls to my main web application and have some code load it based on entries on appsettings.json

was trying to achieve something slimier on Blazor wasm, using some tricks manually i was able to do so, but as the solution grow it wouldn't be efficient to do so specially i want the modules dlls to be marked for lazy loading.

after many hours i was able to come up with this on my main Blazor wasm app csproj
```xml
<Target Name="IncludeMyClientModules" BeforeTargets="_ResolveBlazorWasmOutputs">
    <Message Text="Inside Include Modules Assemblies" Importance="high" />
    <ItemGroup>
      <_BlazorOutputWithTargetPath Include="Modules\FirstModule\FirstModule*.dll"/>
    </ItemGroup>
  </Target>
```

This get the files entries into service-worker-assets.js, and another entry
```xml
<ItemGroup>
    <BlazorWebAssemblyLazyLoad Include="FirstModule.App.dll" />
  </ItemGroup>`
```

Get me an entry into blazor.boot.json lazyAssembly, my two things i need help with are

1- with all this the actual files do not copy to wwwroot\_framework, i tries to do so with couple of other csproj tricks but seems i can't find the right entry point to go before or after.
2- is there a way BlazorWebAssemblyLazyLoad can support patterns, my modules are a couple of dlls like module.app.dll, module.api.client.dll and module.data.models.dll ... so far it seem to get entries on lazyAssembly using BlazorWebAssemblyLazyLoad  i need to list them individually but if i do it like

```xml
<ItemGroup>
    <BlazorWebAssemblyLazyLoad Include="FirstModule.*.dll" />
  </ItemGroup>
```

it doesn't work, any workaround or news that the team intend to support patterns?

**Comment by OnSive** [comment]
// Push
